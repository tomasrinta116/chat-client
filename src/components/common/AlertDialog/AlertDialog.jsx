import React from "react";
import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import DialogTitle from "@material-ui/core/DialogTitle";

const AlertDialog = ({ showAlert, toggleAlert, message, title }) => (
  <React.Fragment>
    <Dialog open={showAlert} onClose={toggleAlert}>
      <DialogTitle>{title}</DialogTitle>
      <DialogContent>
        <DialogContentText className="message-container">
          {message}
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={toggleAlert} color="primary" autoFocus>
          Close
        </Button>
      </DialogActions>
    </Dialog>
  </React.Fragment>
);

export default AlertDialog;