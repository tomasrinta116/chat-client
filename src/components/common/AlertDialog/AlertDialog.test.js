import React from "react";
import { shallow } from "enzyme";
import AlertDialog from './AlertDialog';

describe("<AlertDialog />", () => {

    let wrapper;

    let props = {
        showAlert: true,
        toggleAlert: jest.fn(),
        message: 'this is a test',
        title: 'test message'
    }
    beforeEach(()=>{
        wrapper = shallow(<AlertDialog {...props} />);
    })

    test('it renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
})