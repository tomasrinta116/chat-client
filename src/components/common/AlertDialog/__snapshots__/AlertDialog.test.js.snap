// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertDialog /> it renders correctly 1`] = `
<Fragment>
  <WithStyles(Dialog)
    onClose={[MockFunction]}
    open={true}
  >
    <WithStyles(DialogTitle)>
      test message
    </WithStyles(DialogTitle)>
    <WithStyles(DialogContent)>
      <WithStyles(DialogContentText)
        className="message-container"
      >
        this is a test
      </WithStyles(DialogContentText)>
    </WithStyles(DialogContent)>
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        autoFocus={true}
        color="primary"
        onClick={[MockFunction]}
      >
        Close
      </WithStyles(Button)>
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
</Fragment>
`;
